pkg/billing/client.go

```diff
@@ -40,24 +37,9 @@ func (b *Batch) Count() int {
 	return len(b.events)
 }
 
-func (b *Batch) idempotenize(key string) string {
-	if key != "" {
-		return key
-	}
-
-	return fmt.Sprintf("Host<%s>:ID<%s>:T<%s>", b.c.hostname, uuid.NewString(), time.Now().Format(time.RFC3339))
-}
-
-func (b *Batch) AddAbsoluteEvent(e AbsoluteEvent) {
-	e.Type = "absolute"
-	e.IdempotencyKey = b.idempotenize(e.IdempotencyKey)
-	b.events = append(b.events, &e)
-}
-
-func (b *Batch) AddIncrementalEvent(e IncrementalEvent) {
-	e.Type = "incremental"
-	e.IdempotencyKey = b.idempotenize(e.IdempotencyKey)
-	b.events = append(b.events, &e)
+func (b *Batch) Add(e enrichable) {
```

Comment:
I think exported methods that accept private interfaces (particularly interface types with unexported methods) are a code smell redflag, for me.

I might:
- Make the interface public,
- make batch generic and have it be, basically `type Batch[E IncrementalEvent | AbsoulteEvent] struct {}` and then pass the `E` variable into a method/function that will do the setup, and I think you get all of typing strictness that you want.

Reply:
Made the change. It's... maybe a bit ugly. LMK what you think.

Reply:
I don't see the change?

Reply:
> * Make the interface public,

https://github.com/neondatabase/autoscaling/blob/8579616217587b55150fc1ba578f9bda5b1a4e06/pkg/billing/model.go#L7-L28

> * make batch generic and have it be, basically type Batch[E IncrementalEvent | AbsoulteEvent] struct {} and then pass the E variable into a method/function that will do the setup, and I think you get all of typing strictness that you want.

https://github.com/neondatabase/autoscaling/blob/8579616217587b55150fc1ba578f9bda5b1a4e06/pkg/billing/client.go#L34-L38